src/mongo/db/auth/oidc/oidc_test_fixture.h

```diff
@@ -0,0 +1,203 @@
+
+Copyright (C) 2025-present Percona and/or its affiliates. All rights reserved.
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the Server Side Public License, version 1,
+    as published by MongoDB, Inc.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    Server Side Public License for more details.
+
+    You should have received a copy of the Server Side Public License
+    along with this program. If not, see
+    <http://www.mongodb.com/licensing/server-side-public-license>.
+
+    As a special exception, the copyright holders give permission to link the
+    code of portions of this program with the OpenSSL library under certain
+    conditions as described in each individual source file and distribute
+    linked combinations including the program with the OpenSSL library. You
+    must comply with the Server Side Public License in all respects for
+    all of the code used other than as permitted herein. If you modify file(s)
+    with this exception, you may extend this exception to your version of the
+    file(s), but you are not obligated to do so. If you do not wish to do so,
+    delete this exception statement from your version. If you delete this
+    exception statement from all source files in the program, then also delete
+    it in the license file.
+======= */
+#pragma once
+
+#include "mongo/crypto/jwks_fetcher_factory.h"
+#include "mongo/db/auth/oidc/match_pattern.h"
+#include "mongo/db/auth/oidc/oidc_server_parameters_gen.h"
+#include "mongo/unittest/assert.h"
+#include "mongo/util/periodic_runner.h"
+
+#include <type_traits>
+
+namespace mongo {
+
+static inline bool operator==(const MatchPattern& lhs, const MatchPattern& rhs) {
+    return lhs.toString() == rhs.toString();
+}
+
+// comparator for OidcIdentityProviderConfig objects required by ASSERT_EQ and similar macros
+static inline bool operator==(const OidcIdentityProviderConfig& lhs,
+                              const OidcIdentityProviderConfig& rhs) {
+    return lhs.getIssuer() == rhs.getIssuer() && lhs.getAudience() == rhs.getAudience() &&
+        lhs.getAuthNamePrefix() == rhs.getAuthNamePrefix() &&
+        lhs.getSupportsHumanFlows() == rhs.getSupportsHumanFlows() &&
+        lhs.getJWKSPollSecs() == rhs.getJWKSPollSecs() && lhs.getClientId() == rhs.getClientId() &&
+        lhs.getRequestScopes() == rhs.getRequestScopes() &&
+        lhs.getPrincipalName() == rhs.getPrincipalName() &&
+        lhs.getUseAuthorizationClaim() == rhs.getUseAuthorizationClaim() &&
+        lhs.getAuthorizationClaim() == rhs.getAuthorizationClaim() &&
+        lhs.getLogClaims() == rhs.getLogClaims() && lhs.getMatchPattern() == rhs.getMatchPattern();
+}
+
+// Base class for OidcIdentityProviderConfig's field setters
```

Comment:
This might look slightly overengineered for the current usage, but in my opinion this can be useful when having more unit tests e.g. for the sasl mechanism.